pkg/testutil/remote_config_fake.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil provides utilities that are intended to enable easier
+// and more concise writing of unit test code.
+package testutil
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+)
+
+// FakeRemoteConfig in memory viper implementation of interface `RemoteConfig`.
+type FakeRemoteConfig struct {
+	fileName string
+	v        *viper.Viper
+}
+
+// NewFakeRemoteConfig allocates and returns a new FakeRemoteConfig which is used to reading/writing config stored at location pointed by `fileName`.
+func NewFakeRemoteConfig(str, fileName, configType string) (*FakeRemoteConfig, error) {
```

Comment:
From https://github.com/spf13/viper#reading-config-files, it seems `configType` is optional if the file name has extension?

Reply:
I'm planning to let the fake remote config define the fileName and user only need to provide the configtype